client/state/domains/management/reducer.js

```diff
@@ -72,22 +78,18 @@ export const saving = createReducer( false, {
  * @return {Object}        Updated state
  */
 export const items = createReducer( {}, {
-	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_RECEIVE ]: ( state, { cacheData } ) => ( { ...state, _contactDetailsCache: cacheData } ),
-	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_UPDATE ]:
-		( state, { cacheData } ) =>
-			merge( {}, state, { _contactDetailsCache: cacheData } ),
+	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_RECEIVE ]: ( state, { data } ) => ( { ...state, _contactDetailsCache: data } ),
+	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_UPDATE ]: ( state, { data } ) => {
+		return merge( {}, state, { _contactDetailsCache: data } );
```

Comment:
Ah, right, I didn't think the intention was to merge the contents of `data`. In that case, `merge` can help, though it could lead to sub-branch object mutations that could yield (subtle and pesky) update bugs in React/Redux. Keep that in mind for the future. [react-addons-update](https://facebook.github.io/react/docs/update.html) is our friend here, and already in use in Calypso. I've used in the past to great satisfaction.

It's a legacy lib and we should switch to [immutability-helper](https://github.com/kolodny/immutability-helper), but AFAIK the interface is the same and it should be just a matter of switching the import statements.